a00_qpip/plugin.py

```diff
@@ -123,11 +125,14 @@ def patched_load_plugin(self, packageName):
                 return self._original_loadPlugin(packageName)
             else:
                 log(f"Check on install enabled, we check {packageName}.")
-                self.check_deps_and_prompt_install(additional_plugins=[packageName])
+                dialog, run_gui = self.check_deps(additional_plugins=[packageName])
+                if run_gui:
+                    self.promt_install(dialog)
+                self.save_settings(dialog)
                 self.start_packages([packageName])
                 return True
 
-    def check_deps_and_prompt_install(self, additional_plugins=[], force_gui=False):
+    def check_deps(self, additional_plugins=[]) -> List[MainDialog, bool]:
```

Comment:
Good move in adding type hints !

Can I ask you the final favor of adding a (small) docstring that describes params and return value?

I know we've been bad at this in the past and that I'm asking more than the current state of the code from you here.

Reply:
Sure I'll add some descriptions, do you think that the refactoring splitting it into 3 parts make sense?

Reply:
Yes, it makes sense to me.
I am wondering if we could split some things away into a non-gui part (e.g. there are QGIS server plugins too, we might want to introduce a command line interface for this one day), but I did not check how big this kind of refactoring would be.

Reply:
ok, but that sounds like another PR :) Let me know if there is anything else to fix in this PR and if there is anything on your mind to make the plugin none experimental.

Reply:
Left it as #13